absl/strings/internal/char_map.h

```diff
@@ -32,7 +32,7 @@ namespace strings_internal {
 
 class Charmap {
  public:
-  constexpr Charmap() : m_() {}
+  Charmap() = default;
```

Comment:
`m_` is an array whose default construction is trivial, whereas `m_()` will do value-initialization setting it to zero. 

https://godbolt.org/z/zhZjwr

But perhaps `uint64_t m_[4] = {};` would enable this change.

Reply:
Apart from gaining the correct exception specification, (which we could simply add), I'm not sure this change provides any benefit.

Reply:
Quite frankly it depends.

1. You have to do work to get the same behavior 
2. As far as I know virtually everyone and their mother agree that member initialization if preferable to initialization to a default value in the constructor.
3. It is more consitent. There are a lot of places in abseil where implicitely defaulted member functions are used.

@asoffer You are correct. However, I am not sure whether member initialization is valid in C++11 mode?

Reply:
Member initialization is valid but I think we still want `constexpr` default cosntructors and so `= default` is probably not what we want. I agree with your sentiment that this is a deficiency in the standard and would love to see your paper.

Reply:
Hm I started to write it and idway through it i thought about actually reading the relevant pages of the standard. (Perfect ordering...)

It seems that I get a pass this time because of Section 11.4.2 Explicitly-defaulted functions [dcl.fct.def.default] paragraph 2:

> An explicitly-defaulted function that is not deﬁned as deleted may be declared constexpr only if it would have been implicitly declared as constexpr. If a function is explicitly defaulted on its ﬁrst declaration, it is implicitly considered to be constexpr if the implicit declaration would be.

So according to the standard we should already be able to use `constexpr foo() = default;` I am 95% sure that i tried it and ended up with all hell breaking loose but I will try it again.